src/marks/axis.js

```diff
@@ -81,9 +81,9 @@ function axisKy(
     x,
     margin,
     marginTop = margin === undefined ? 20 : margin,
-    marginRight = margin === undefined ? (anchor === "right" ? 40 : 0) : margin,
+    marginRight = margin === undefined ? (anchor === "right" ? (x == null ? k : 40) : 0) : margin,
```

Comment:
Temporarily setting this to `k` feels a bit dirty… it’s effectively a hidden API if you set the marginRight option to "x" or "y", right?

Reply:
Agree, but I didn't find a better way.

Reply:
You’ll likely have to change how the defaults are initialized rather than using default initializers.